### PR TITLE
JENKINS-51291 Make new view link relative

### PR DESCRIPTION
**Pull request by Austry**
See [JENKINS-51291](https://issues.jenkins-ci.org/browse/JENKINS-51291).

### Proposed changelog entries
"New view" absolute link changed to relative

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change).
- [x] Appropriate autotests or explanation to why this change has no tests

### CODE REVIEW

**Comment by daniel-beck** [comment]
Supersedes #3633… @Wadeck might know whether this is a viable solution (it's probably not, looking at the endless discussion over there).

File: core/src/main/java/jenkins/model/NewViewLink.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public String getDisplayName() {
 
             @Override
             public String getUrlName() {
-                return Jenkins.get().getRootUrl() + URL_NAME;
+                return URL_NAME;
```
Comment (Wadeck) [inline core/src/main/java/jenkins/model/NewViewLink.java]:
```suggestion
                return "/" + URL_NAME;
```

The relative view will not work within an already existing view as mentioned by https://github.com/jenkinsci/jenkins/pull/3178. It's the case for regular "List View" (Core). For certain view, like Nested View (plugin), the feature is expected and it adds itself the action button to create new View inside the nested view, resulting in duplicate (no change with your PR).

That fix was incorrect for the proxy URL that you are trying to solve, but with your solution you are going to introduce previous problem back.

Multiple ways to improve the situation without coming back with regression:
1) Use "absolute" relative link, meaning, it's relative to the contextPath, but absolute *inside* Jenkins. If you are in a folder or in a view, the action will propose a new view "globally". This is a bit counter intuitive in terms of UX. (that's the suggestion, the easiest way but not perfect one)
2) Add support to create view from all the views (could be tricky)
3) Do not display the generic new View, if the current ancestor is a view.

Concerning the point 3, perhaps something to do on this line: [NewViewLink.java#L25](https://github.com/jenkinsci/jenkins/blob/0e62b68012282648377cfac4fe6d01956d7c1f11/core/src/main/java/jenkins/model/NewViewLink.java#L25). At this point, if you are in a `ListView`, we should not display the link as ListView does not support nested view. If we are in a folder or in Jenkins home page, the v is an instance of `AllView`, and the link will be to `AbstractFolder/newView.jelly` or `Jenkins/newView.jelly` respectively, leading once posted, to `AbstractFolder#doCreateNewView` or `Jenkins#doCreateNewView`. There are perhaps mechanisms that I am not aware to inform that "you" can accept new view creation or similar stuff.

I would recommend to just go with proposal 1 and in a follow-up PR to look at other solutions.

**Review by Wadeck — CHANGES_REQUESTED** [review]
The current proposal will cause a regression of [JENKINS-48447](https://issues.jenkins-ci.org/browse/JENKINS-48447).

**Comment by jglick** [comment]
> The current proposal will cause a regression of [JENKINS-48447](https://issues.jenkins-ci.org/browse/JENKINS-48447).

Yes, but it will fix other problems—not just the originally reported issue with proxy servers (which would be worked around merely by setting the Jenkins root URL correctly), but also operation inside folders (no workaround).

See rule 1 in the [guide](https://wiki.jenkins.io/display/JENKINS/Hyperlinks+in+HTML).

**Comment by varyvol** [comment]
@Austry @Wadeck @jglick is there any other viable solution to fix this without causing a regression? If there are ideas, maybe that would help moving this forward.

**Comment by jglick** [comment]
> is there any other viable solution to fix this without causing a regression?

Probably. I do not know offhand.

**Comment by alecharp** [comment]
This seems to be stuck, so I'm proposing to close this PR.

**Comment by jglick** [comment]
Not sure if this PR is right; only that #3178 (the current state) is definitely wrong.

**Comment by varyvol** [comment]
Closing given there's been no activity or answer. Please let us know in case you want to work on this again.

**Pull request closed without merging**